docs/e2e-test.md

```diff
@@ -0,0 +1,44 @@
+# Ramen End-to-End (E2E) Test Framework
+
+This document will provide an introduction to the framework, describe its components and structure, and explain how to run the tests.
+
+## Overview
+
+The E2E test framework is designed to validate the functionality of the Ramen project from a user's perspective. This means that it will test the entire system. The framework is built using Go and follows the native Go test format. Setting up the test environment is outside the scope of the framework, and is left to the user. The framework will however, detect the environment and run the tests accordingly.
```

Comment:
Very hard to read when lines are not wrapped. Please use short lines (72 chars works good for text).